overlord/servicestate/servicemgr.go

```diff
@@ -42,8 +53,215 @@ func Manager(st *state.State, runner *state.TaskRunner) *ServiceManager {
 	return m
 }
 
+func MockEnsureSnapServices(mgr *ServiceManager, ensured bool) (restore func()) {
+	old := mgr.ensuredSnapSvcs
+	mgr.ensuredSnapSvcs = ensured
+	return func() {
+		mgr.ensuredSnapSvcs = old
+	}
+}
+
+func restartServicesKilledInSnapdRefresh(modified map[*snap.Info][]*snap.AppInfo) error {
+	// we decide on which services to restart by identifying (out of the set of
+	// services we just modified) services that were stopped after
+	// usr-lib-snapd.mount was written, but before usr-lib-snapd.mount was last
+	// stopped - this is the time window in which snapd (accidentally) killed
+	// all snap services using Requires=, see LP #1924805 for full details, so
+	// we need to undo that by restarting those snaps
+
+	// TODO: use the var from core18.go here instead
+	st, err := os.Stat(filepath.Join(dirs.SnapServicesDir, wrappers.SnapdToolingMountUnit))
+	if err != nil {
```

Comment:
Do we need to consider os.IsNotExist ? Or will this only run on systems that have this unit?

Reply:
AFAIU this is run on core18+ only, as such maybe the func should be named restartServicesKilledInSnapdSnapRefresh ?

Reply:
this function indeed should only be run on UC18+, so we will always have this unit around on a proper install. I can adjust the name of the function as Maciej suggested to make this more clear